src/generic/grid.cpp

```diff
@@ -4551,92 +4551,90 @@ wxGrid::DoGridCellLeftDown(wxMouseEvent& event,
         return;
     }
 
+    // Clicking on (or very near) the separating lines shouldn't change the
+    // selection when it's used for resizing -- but should still do it if
+    // resizing is disabled (notice that we intentionally don't check for
+    // it being disabled for a particular row/column as it would be
+    // surprising to have different mouse behaviour in different parts of
+    // the same grid, so we only check for it being globally disabled).
+    int dragRowOrCol = wxNOT_FOUND;
+    if ( CanDragGridColEdges() )
+        dragRowOrCol = XToEdgeOfCol(pos.x);
+
+    if ( dragRowOrCol == wxNOT_FOUND && CanDragGridRowEdges() )
+        dragRowOrCol = YToEdgeOfRow(pos.y);
+
+    if ( dragRowOrCol != wxNOT_FOUND )
+    {
+        DoStartResizeRowOrCol(dragRowOrCol);
+        return;
+    }
+
+    DisableCellEditControl();
+    MakeCellVisible( coords );
+
     if ( event.ShiftDown() && !event.CmdDown() )
     {
         if ( m_selection )
         {
             m_selection->ExtendCurrentBlock(m_currentCellCoords, coords, event);
             MakeCellVisible(coords);
         }
-    }
-    else
+        // Return to don't set `coords` cell current.
+        return;
+    } else if ( event.CmdDown() && !event.ShiftDown() )
```

Comment:
I'd like to avoid returning from the middle of this function (yes, I know that there are already existing `return`s there, but they're simpler than this one), so could we rather do:
```suggestion
    }
    else
    {
        if ( event.CmdDown() && !event.ShiftDown() )
```
and keep `SetCurrentCell(coords)` inside that `else`?